src/StubExecutable/StubExecutable.cpp

```diff
@@ -40,6 +40,79 @@ wchar_t* FindOwnExecutableName()
 	return ret;
 }
 
+// Credit to https://stackoverflow.com/a/35717/3193009
+LONG GetStringRegKey(HKEY hKey, const std::wstring &strValueName, std::wstring &strValue)
+{
+    WCHAR szBuffer[512];
+    DWORD dwBufferSize = sizeof(szBuffer);
+    ULONG nError;
+    nError = RegQueryValueExW(hKey, strValueName.c_str(), 0, NULL, (LPBYTE)szBuffer, &dwBufferSize);
+    if (ERROR_SUCCESS == nError)
+    {
+        strValue = szBuffer;
+    }
+    return nError;
+}
+
+// Credit to https://stackoverflow.com/a/6218957/3193009
+BOOL FileExists(LPCTSTR szPath)
+{
+  DWORD dwAttrib = GetFileAttributes(szPath);
+
+  return (dwAttrib != INVALID_FILE_ATTRIBUTES &&
+         !(dwAttrib & FILE_ATTRIBUTE_DIRECTORY));
+}
+
+std::wstring GetBraveCorePath(HKEY hKey, LPCTSTR lpSubKey)
+{
+	HKEY hBraveKey;
+	std::wstring fullPath;
+	if (ERROR_SUCCESS == RegOpenKeyExW(hKey,
+		lpSubKey, 0, KEY_READ, &hBraveKey))
+	{
+		if (ERROR_SUCCESS == GetStringRegKey(hBraveKey, L"path", fullPath))
+		{
+			size_t path_index = fullPath.find(L"Update\\BraveUpdate.exe");
```

Comment:
Are we going to support upgrading Beta browser-laptop to Beta channel brave-browser? If so, the reg key and value are still the same, but the exe path is at a different location.

In other words, when a Windows user only has the beta version of brave-browser installed, then the update path is still at:
`C:\Program Files (x86)\BraveSoftware\Update\BraveUpdate.exe`
but the exe path is at
`C:\Program Files (x86)\BraveSoftware\Brave-Browser-Beta\Application\brave.exe`

If we're going to support it, we may want to parameterize the channel or figure out the path another way.

Reply:
Good question - at the moment, I'm only checking for (and installing) the stable Brave Browser

For complexity sake, I think it might be best to stick to this. WDYT, @rebron?

Reply:
Let me run through the approvals but getting all muon users (regardless of muon channel - nightly/dev/beta/release) should all go to release/stable Brave Browser makes sense. 

Rational for keeping muon users in the same/corresponding channel was their initial choice of the channel and for testing purposes. If there's little value in the latter, let's stick with what you have and keep things simple and get everyone onto the brave-core release channel.